### ISSUE TITLE
[WTEXport] Export fails if input files are read-only

### ISSUE THREAD

**Issue by biwa** [MTrop/DoomTools#46]
WTEXport fails to export if either the resources you want to export from, or the base (I)WAD is set to read-only.

For example my doom2.wad is set to read only, using it as a base results in the following error:

```
>wtexscan MAP01.wad | wtexport.cmd ..\..\wads\resource\OTEX_1.1.WAD --base-wad ..\..\iwads\doom2.wad --output mapstex.wad --create
Read texture/flat list...
Scanning ..\..\iwads\doom2.wad...
ERROR: "..\..\iwads\doom2.wad" could not be read.
```

When removing the read-only flag the export works as expeced.

This also happens when the resource to export from is set as read-only, although it bails out later:

```
>wtexscan MAP01.wad | wtexport.cmd ..\..\wads\resource\OTEX_1.1.WAD --base-wad ..\..\iwads\doom2.wad --output mapstex.wad --create
Read texture/flat list...
Scanning ..\..\iwads\doom2.wad...
    Scanning TEXTUREx/PNAMES...
        428 entries in TEXTURE1.
        469 entries in PNAMES.
    Scanning patch entries...
        475 patches.
    Scanning flat entries...
        153 flats.
    Scanning texture namespace entries...
        0 namespace textures.
Scanning ..\..\wads\resource\OTEX_1.1.WAD...
ERROR: "..\..\wads\resource\OTEX_1.1.WAD" could not be read.
```

**Comment by MTrop** [comment]
It might be because of the way that the main library ([DoomStruct](https://github.com/MTrop/DoomStruct)) opens WAD files for reading, since it opens them for random access in `rws` mode. The OS is probably attempting to acquire a write channel and failing (and rightfully so).

https://github.com/MTrop/DoomStruct/blob/9f8b6875159fc15bb655b927cc2ee0647f42b11e/src/main/java/net/mtrop/doom/WadFile.java#L94

A mode may need to be made for just forcing a read-only mode on WADs that I know won't be written to - a big oversight on my part! This means that a bunch of stuff in DoomTools may need tweaking afterward to accommodate.

**Comment by MTrop** [comment]
Heads up: I may not need to change much of anything due to how I'm implementing the fix in DoomStruct, but this issue is still going to kept open until the new version of DoomStruct that fixes this is added to the build.

**Comment by MTrop** [comment]
Fixed in commit e586cbba50cd3af6dd03fc066b4fe7885c9c0d49 with the updating of DoomStruct to version 2.14.0.
